cluster/gce/trusty/configure.sh

```diff
@@ -22,7 +22,8 @@
 
 config_hostname() {
   # Set the hostname to the short version.
-  short_hostname=$(hostname -s)
+  host_name=$(curl --silent "http://metadata.google.internal/computeMetadata/v1/instance/hostname" -H "Metadata-Flavor: Google")
```

Comment:
The only issue is that this gives you a name with the `.c.<project_name>.internal` suffix. I think in our other scripts we get a "short" hostname without the domain.